external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutConfig.java

```diff
@@ -79,16 +78,46 @@
         UNCOMMITTED_EARLIEST,
         UNCOMMITTED_LATEST }
     
+    /**
+     * Convenience method to get a Builder for String key/value spouts. Sets the key/value Deserializers to the StringDeserializer class.
+     * Users who need different deserializers should use the Builder constructors instead, 
+     * and set the deserializer classes via {@link Builder#setProp(java.lang.String, java.lang.Object)}
+     * @param bootstrapServers The bootstrap servers for the consumer
+     * @param topics The topics to subscribe to
+     * @return The new builder
+     */
     public static Builder<String, String> builder(String bootstrapServers, String ... topics) {
-        return new Builder<>(bootstrapServers, StringDeserializer.class, StringDeserializer.class, topics);
+        return setDefaultStringDeserializers(new Builder<>(bootstrapServers, topics));
     }
     
+    /**
+     * Convenience method to get a Builder for String key/value spouts. Sets the key/value Deserializers to the StringDeserializer class.
+     * Users who need different deserializers should use the Builder constructors instead, 
+     * and set the deserializer classes via {@link Builder#setProp(java.lang.String, java.lang.Object)}
+     * @param bootstrapServers The bootstrap servers for the consumer
+     * @param topics The topics to subscribe to
+     * @return The new builder
+     */
     public static Builder<String, String> builder(String bootstrapServers, Collection<String> topics) {
-        return new Builder<>(bootstrapServers, StringDeserializer.class, StringDeserializer.class, topics);
+        return setDefaultStringDeserializers(new Builder<>(bootstrapServers, topics));
     }
     
+    /**
+     * Convenience method to get a Builder for String key/value spouts. Sets the key/value Deserializers to the StringDeserializer class.
+     * Users who need different deserializers should use the Builder constructors instead, 
+     * and set the deserializer classes via {@link Builder#setProp(java.lang.String, java.lang.Object)}
+     * @param bootstrapServers The bootstrap servers for the consumer
+     * @param topics The topic pattern to subscribe to
+     * @return The new builder
+     */
     public static Builder<String, String> builder(String bootstrapServers, Pattern topics) {
-        return new Builder<>(bootstrapServers, StringDeserializer.class, StringDeserializer.class, topics);
+        return setDefaultStringDeserializers(new Builder<>(bootstrapServers, topics));
+    }
+    
+    private static Builder<String, String> setDefaultStringDeserializers(Builder<String, String> builder) {
+        builder.setProp(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
```

Comment:
This is just here to support the 3 convenience builder() functions above. I'll rename it so it doesn't imply that String is the default type.